content/en/docs/concepts/workloads/pods/ephemeral-containers.md

```diff
@@ -0,0 +1,219 @@
+clusters. You should expect the feature not to work in some situations, like
+when targeting the namespaces of a container. In accordance with the [Kubernetes
+Deprecation Policy](/docs/reference/using-api/deprecation-policy/), this alpha
+feature could change significantly in the future or be removed entirely.
+{{< /warning >}}
+
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Understanding ephemeral containers
+
+{{< glossary_tooltip text="Pods" term_id="pod" >}} are the fundamental building
+block of Kubernetes applications. Since pods are intended to be disposable and
+replaceable, you cannot add a container to a pod once it has been created.
+Instead, you usually delete and replace pods in a controlled fashion using
+[deployments](/docs/concepts/workloads/controllers/deployment/).
+
+Sometimes it's necessary to inspect the state of an existing pod, however, for
+example to troubleshoot a hard-to-reproduce bug. In these cases you can run
+an ephemeral container in an existing pod to inspect its state and run
+arbitrary commands.
+
+### Differences from regular containers
+
+Ephemeral containers lack guarantees for resources or execution, and they will
+never be automatically restarted, so they are not appropriate for building
+applications.  Ephemeral containers are described using the same `ContainerSpec`
+as regular containers, but many fields are incompatible and disallowed for
+ephemeral containers.
+
+- Ephemeral containers may not have ports, so fields such as `ports`,
+  `livenessProbe`, `readinessProbe` are disallowed.
+- Pod resource allocations are immutable, so setting `resources` is disallowed.
+- For a complete list of allowed fields, see the [EphemeralContainer reference
+  documentation](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#ephemeralcontainer-v1-core).
+
+Ephemeral containers are created using a special `ephemeralcontainers` handler
+in the API rather than by adding them directly to `pod.spec`, so it's not
+possible to add an ephemeral container using `kubectl edit`.
+
+Like regular containers, ephemeral containers may not be changed or removed
+once they are added to the pod.
+
+## Uses for ephemeral containers
+
+Ephemeral containers are useful for interactive troubleshooting when `kubectl
+exec` is insufficient because a container has crashed or a container image
+doesn't include debugging utilities.
+
+In particular, [distroless images](https://github.com/GoogleContainerTools/distroless)
+enable you to deploy minimal container images that reduce attack surface
+and exposure to [CVEs](https://cve.mitre.org/). Since they do not include a
+shell or any debugging utilities, it's difficult to troubleshoot distroless
+images using `kubectl exec` alone.
+
+When using ephemeral containers, it's helpful to enable [process namespace
+sharing](/docs/tasks/configure-pod-container/share-process-namespace/) so
+you can view processes in other containers.
+
+### Examples
+
+{{< caution >}}
```

Comment:
I would use `{{< note >}}` shortcode here as there's no risk to warn readers about.

Reply:
I started with note, but then noticed that the style guide suggested note was for tips and caution was to avoid pitfalls. This seemed like a pitfall since nothing will work if the feature is disabled.

I don't feel strongly, so happy to do whatever reviewers suggest.

Reply:
Go for `{{< note >}}`; you can't break your cluster by getting this wrong (I believe?)